File: Rado/part365/src/app/model/part/partView.ts
import { Part } from "./part";

export class PartView extends Part{
    companyName?     : string;
    companyId?       : number;
    modelName?       : string;
    modificationName?: string;
    regNumber?       : string;
    vin?             : string;
    engineTypeDesc?  : string;
    categoryName?    : string;
    createdTime?     : number;
    sellerName?      : string;
    sellerPhone?     : string;
    sellerPhone2?    : string
    sellerViber?     : string
    sellerWebPage?   : string;
    numberImages?    : number;
    yearName?        : string;
}

File: Models/Models/PartView.cs
using Models.Models;
using System.Text.Json.Serialization;

namespace Rado.Models
{
    public class PartView : Part
    {
        public string CompanyName { get; set; }
        public int CompanyId { get; set; }
        public string ModelName { get; set; }
        public string ModificationName { get; set; }
        public string CategoryName { get; set; }
        public bool IsCar { get; set; }
        public string RegNumber { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string Vin { get; set; }
        public int NumberImages { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string YearName { get; set; }
        public ImageDataClass MainImageDataClass { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public TraderDetails TraderDetails { get; set;}

        public PartView()
        {
            LeftRightPosition = FrontBackPosition = 0;
        }

    }
}
